### PR TITLE
London | May-2025 | KhilolaRustamova |Mdg/sprint 2

### PR DESCRIPTION
**Pull request by HilolaRustam**
## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the requirements of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

### CODE REVIEW

File: Sprint-2/debug/recipe.js
Comment (AdnanGondal) [inline Sprint-2/debug/recipe.js:1]: Great explanations and solutions for the debug section - well done.

File: Sprint-2/implement/lookup.test.js
Comment (AdnanGondal) [inline Sprint-2/implement/lookup.test.js:1]: Perhaps it would be better to add different edge cases to these tests such as what happens when:
- Empty input
- When there are duplicate keys
- When a pair has extra values, such as createLookup([["UK", "GBP", "Extra"]])).

File: Sprint-2/implement/querystring.test.js
Comment (AdnanGondal) [inline Sprint-2/implement/querystring.test.js:1]: A great set of test cases. Just had one question. 
What do you expect to happen when you run parseQueryString("a=1&&b=2").
Can you add a test case for it?

File: Sprint-2/implement/querystring.js [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
+  if (!queryString) return queryParams;
+  
   const keyValuePairs = queryString.split("&");
 
   for (const pair of keyValuePairs) {
-    const [key, value] = pair.split("=");
-    queryParams[key] = value;
-  }
+   const [key, ...rest] = pair.split("="); // Grab everything after first '='
+   const value = rest.join("="); // Safely join back if value had '=' signs
```
[the remark this answers is not in the document]
Comment (HilolaRustam) [reply Sprint-2/implement/querystring.js:9]: i did, thank you
